### ISSUE TITLE
Inconsistent enablement of Federation

### ISSUE THREAD

**Issue by t1** [smallrye/smallrye-graphql#1689]
In 1.9.1, enabling Federation is not automatic any more and is only possible with a system property

**Comment by jmartisk** [comment]
No, it's automatic, at least it will be when quarkus updates to SmallRye GraphQL 1.9.1 via this PR: https://github.com/quarkusio/quarkus/pull/30227/files

For WildFly, I am going to do something similar.

**Comment by jmartisk** [comment]
I want to make the automatic enablement handled by the runtime, not directly in SmallRye, because different runtimes have different configuration mechanisms - for example Quarkus has its own configuration property for enabling federation (`quarkus.smallrye-graphql.federation.enabled`), and I want Quarkus to decide based on THAT property, not based only on `smallrye.graphql.federation.enabled`

**Comment by t1** [comment]
Okay, I see... but wouldn't it make more sense to implement it once? Have you seen #1690

**Comment by jmartisk** [comment]
I've commented on the PR, I think it might make sense, but it would need some changes, let's move the discussion there
